### ISSUE TITLE
tiff fd leak

### ISSUE THREAD

**Issue by lambdafu** [python-pillow/Pillow#2181]
### What did you do?

```
PYTHONPATH=. python fdleaktest.py
```

with fdleaktest.py:

``` python
from PIL import Image
d={}
for i in range(2048):
    try:
        d[i] = Image.open("Tests/images/multipage-lastframe.tif")
        d[i].load()
        d[i].close()
    except IOError as e:
        print "Failed to open %i files" % i
        raise e
```
### What did you expect to happen?

That the script completes without error.
### What actually happened?

```
Failed to open 1021 files
Traceback (most recent call last):
  File "../fdleak/test-3.py", line 10, in <module>
IOError: [Errno 24] Too many open files: 'Tests/images/multipage-lastframe.tif'
```
### What versions of Pillow and Python are you using?

master branch abc4d55f08a61be8e1fb7815b56423c0dff268c1 and Python 2.7.12

Note: png (for example) works fine.

Note: Under Windows, this prevents such image files from being removed (because Windows does not allow to unlink opened files).

**Comment by lambdafu** [comment]
Also works with TiffImagePlugin.READ_LIBTIFF = True

**Comment by lambdafu** [comment]
Could be as simple as this in TiffImageFile:
```
    def close(self):
        if self.__fp:
            self.__fp =   None
        super(TiffImageFile, self).close()
```
Will make pull request if I can confirm this on Windows with more testing.

**Comment by wiredfool** [comment]
fixed in #2330
